### PR TITLE
Rule S4058: Overloads with a 'StringComparison' parameter should be used

### LINKED ISSUE #509: Rule S4058: Overloads with a "StringComparison" parameter should be used
[RSPEC-4058](https://jira.sonarsource.com/browse/RSPEC-4058)

Many string operations, the Compare and Equals methods in particular, provide an overload that accepts a StringComparison enumeration value as a parameter. Calling these overloads and explicitly providing this parameter makes your code clearer and easier to maintain.

Contributes to [MMF-963](https://jira.sonarsource.com/browse/MMF-963).

### PR DESCRIPTION
**Pull request by Evangelink**
Fix #509

### CODE REVIEW

File: sonaranalyzer-dotnet/its/expected/Ember-MM/Ember.Plugins-{9496C697-5AFD-4813-AEDC-AF33FACEADF0}-S4058.json [hunk 1]
```diff
@@ -0,0 +1,69 @@
+{
+"issues":  [
+{
+"id":  "S4058",
+"message":  "Change this call to 'lTitle.EndsWith' to an overload that accepts a 'StringComparison' as a parameter.",
```
Comment (Evangelink) [inline sonaranalyzer-dotnet/its/expected/Ember-MM/Ember.Plugins-{9496C697-5AFD-4813-AEDC-AF33FACEADF0}-S4058.json]: I am mentioning the full expression of the invocation, maybe it should be reduced, WDYT?

Comment (michalb-sonar) [inline sonaranalyzer-dotnet/its/expected/Ember-MM/Ember.Plugins-{9496C697-5AFD-4813-AEDC-AF33FACEADF0}-S4058.json]: I like it, as it gives more context. But it's not what we normally do so it might be confusing in some cases and won't be consistent. In this very case, I'm fine with either.

Perhaps we should start adding full expression in all messages? @Evangelink  @valhristov @jccollet WDYT?

File: sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/RspecStrings.resx [hunk 2]
```diff
@@ -151,7 +151,7 @@
     <value>Default arguments are determined by the static type of the object. If a default argument is different for a parameter in an overriding method, the value used in the call will be different when calls are made via the base or derived object, which may be contrary to developer expectations. </value>
   </data>
   <data name="S1006_IsActivatedByDefault" xml:space="preserve">
-    <value>True</value>
+    <value>False</value>
```
Comment (michalb-sonar) [inline sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/RspecStrings.resx]: These changes to resx file should not be here.

Comment (Evangelink) [inline sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/RspecStrings.resx]: Ah ffs I don't understand why these changes keep popping-up!
